plugins/reindex/src/main/java/org/elasticsearch/plugin/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -179,10 +132,15 @@ public void onFailure(Throwable e) {
     void onScrollResponse(SearchResponse searchResponse) {
         scroll.set(searchResponse.getScrollId());
         if (searchResponse.getShardFailures() != null && searchResponse.getShardFailures().length > 0) {
-            Collections.addAll(searchFailures, searchResponse.getShardFailures());
+            task.addSearchFailures(searchResponse.getShardFailures());
             startNormalTermination();
             return;
         }
+        long total = searchResponse.getHits().totalHits();
+        if (mainRequest.getSize() >= 0) {
```

Comment:
We do! Over in `AbstractBulkByScrollRequest` we do:
```java
if (false == (size == -1 || size > 0)) {
   e = addValidationError(
```

I'll drop the `=` from the `>=` because it is confusing.

In the case where the scroll request returns 0 results we'll be aborting the request so early that I don't think the ambiguity is a big deal there.